building python3Packages.protobuf3_2 on OS X fails due to string include

## Issue description

Building the python 3 protobuf package fails for me on macOS.

```
$ nix-build '<nixpkgs>' -A python3Packages.protobuf3_2
...
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I. -I../src -I/nix/store/nkmdx27mxqq4mz9qvnbd8v754y290ndf-python3-3.6.1/include/python3.6m -c google/protobuf/pyext/descriptor.cc -o build/temp.macosx-10.6-x86_64-3.6/google/protobuf/pyext/descriptor.o -Wno-write-strings -Wno-invalid-offsetof -Wno-sign-compare -Wno-shorten-64-to-32 -std=c++11
clang-4.0: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
google/protobuf/pyext/descriptor.cc:35:10: fatal error: 'string' file not found
#include <string>
         ^~~~~~~~
1 error generated.
error: command 'clang' failed with exit status 1
builder for ‘/nix/store/2z36gcrnmi2bz1a3s3nppc8imp11ppgj-python3.6-protobuf-3.2.0.drv’ failed with exit code 1
error: build of ‘/nix/store/2z36gcrnmi2bz1a3s3nppc8imp11ppgj-python3.6-protobuf-3.2.0.drv’ failed
```

Digging in (by launching ` nix-shell -A python3Packages.protobuf3_2`), the cause seems to be that `python setup.py build_ext --cpp_implementation` calls `CC=clang` instead of `CXX=clang++`. The `clang` call above fails, but the same call works after replacing `clang` by `clang++`. Similarly, `CC=$CXX python setup.py build_ext --cpp_implementation` works fine.

That in turn matters because of

```
if [[ "$isCpp" = 1 ]]; then
    if [[ "$cppInclude" = 1 ]]; then
        NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE ${NIX_CXXSTDLIB_COMPILE-}"
    fi
    NIX_CFLAGS_LINK="$NIX_CFLAGS_LINK $NIX_CXXSTDLIB_LINK"
fi
```

in the `clang`/`clang++` wrappers, because the system include path `NIX_CXXSTDLIB_COMPILE` is missing in the `clang` case.
## Technical details

* System: mac OS 10.12.5
* Nix version: nix-env (Nix) 1.11.8
* Nixpkgs version: "17.09pre108299.ec9a23332f"
* Sandboxing enabled: grep: /etc/nix/nix.conf: No such file or directory :)

When digging in, I used a checkout of nixpkgs with hash 63e9d1c51ec0ad41c3ef729851442f199ce06a25, but it seems to make no difference.

--- a/pkgs/development/python-modules/protobuf.nix
+++ b/pkgs/development/python-modules/protobuf.nix
@@ -1,5 +1,5 @@
 { stdenv, python, buildPythonPackage
-, protobuf, google_apputils, pyext
+, protobuf, google_apputils, pyext, libcxx
 , disabled, doCheck ? true }:
 
 with stdenv.lib;
@@ -7,6 +7,9 @@ with stdenv.lib;
 buildPythonPackage rec {
   inherit (protobuf) name src;
   inherit disabled doCheck;
+
+  # work around python distutils compiling C++ with $CC
+  NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin "-I${libcxx}/include/c++/v1";
 
   propagatedBuildInputs = [ protobuf google_apputils ];
   buildInputs = [ google_apputils pyext ];

